src/NServiceBus.NHibernate/Outbox/OutboxPersister.cs

```diff
@@ -75,10 +83,11 @@ public void SetAsDispatched(string messageId)
                 {
                     using (var tx = session.BeginTransaction(IsolationLevel.ReadCommitted))
                     {
-                        var queryString = string.Format("update {0} set Dispatched = true, DispatchedAt = :date where MessageId = :messageid And Dispatched = false",
+                        var queryString = string.Format("update {0} set Dispatched = true, DispatchedAt = :date where MessageId IN ( :messageid, :qualifiedMsgId ) And Dispatched = false",
                             typeof(OutboxRecord));
                         session.CreateQuery(queryString)
                             .SetString("messageid", messageId)
+                            .SetString("qualifiedMsgId", EndpointQualifiedMessageId(messageId))
```

Comment:
`messageid` and `...MsgId`. the OCD it burns  

![](http://i.ytimg.com/vi/vLvo888gY-Q/maxresdefault.jpg)